x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -300,6 +302,27 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
                 wrapperWarnings.push(warningMissingTimestampFieldsMessage);
               }
               skipExecution = foundNoIndices;
+
+              const frozenFieldCaps = await services.scopedClusterClient.asCurrentUser.fieldCaps({
```

Comment:
should we avoid these calls on Serverless?

Reply:
Yeah, I added an `isServerless` flag for rule executors so we can tweak rule execution as needed for the different environments. We don't have any precedent for that in rule execution yet right?

Reply:
we have one that calls for data tier filters from advanced settings - https://github.com/elastic/kibana/blob/8.18/x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/get_data_tier_filter.ts

getting advanced settings does not seen too performance affecting, but we something we can improve using this setting